### PR TITLE
Add Save and Load methods to persist seed in local file system

### PR DESCRIPTION
**Pull request by jazz-cb**
### What changed? Why?
Adding a save and load method to users class that can be used to optionally persist the seeds.

Usage:

To save without encryption on
```
u = Coinbase.default_user
w = u.create_wallet
u.save(w)
wallets=u.load()
wallets[w.wallet_id].wallet_id == w.wallet_id
```

To save with encryption on
```
u.save(w)
wallets=u.load()
wallets[w.wallet_id].wallet_id == w.wallet_id
```

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -128,15 +128,33 @@ In order to persist the data for the Wallet, you will need to implement a store
 # the data required to re-instantiate the Wallet at a later time.
 store(data)
 ```
+For convenience during testing, we provide a save method that stores the Wallet data in your local file system.
+This is an insecure method of storing your keys and should only be used for development purposes.
+```ruby
+u.save(w3)
+```
+
+To encrypt the saved data, set encrypt_flag to true.
```
Comment (yuga-cb) [inline README.md:137]: set encrypt to true?

File: README.md [hunk 2]
```diff
@@ -128,15 +128,33 @@ In order to persist the data for the Wallet, you will need to implement a store
...
 The below code demonstrates how to re-instantiate a Wallet from the data export.
 
 ```ruby
 # The Wallet can be re-instantiated using the exported data.
-# w2 will be equivalent to w1.
+# w4 will be equivalent to w3.
 w4 = u.import_wallet(data)
 ```
 
+To import wallets that were persisted to local file system using `save()`, use the below code.
```
Comment (yuga-cb) [inline README.md:150]: persisted to your local file system

File: README.md [hunk 3]
```diff
@@ -128,15 +128,33 @@ In order to persist the data for the Wallet, you will need to implement a store
...
+```ruby
+# The Wallet can be re-instantiated using the exported data.
+# w5 will be equivalent to w3.
+wallets = u.load()
```
Comment (yuga-cb) [inline README.md:154]: No need for ()

File: lib/coinbase/user.rb [hunk 4]
```diff
@@ -62,6 +62,65 @@ def list_wallet_ids
       wallets.data.map(&:id)
     end
 
+    # Saves a wallet to local file system. Wallet saved this way can be re-instantiated with load() function,
+    # provided the backup_file is available.
+    def save(wallet, encrypt_flag = false)
```
Comment (yuga-cb) [inline lib/coinbase/user.rb:67]: Document params

Also just encrypt, as a named parameter (encrypt: )

File: lib/coinbase/user.rb [hunk 5]
```diff
@@ -62,6 +62,65 @@ def list_wallet_ids
       wallets.data.map(&:id)
     end
 
+    # Saves a wallet to local file system. Wallet saved this way can be re-instantiated with load() function,
+    # provided the backup_file is available.
```
Comment (yuga-cb) [inline lib/coinbase/user.rb:66]: Add comment saying this is only for testing purposes, it's insecure, etc.

File: lib/coinbase/user.rb [hunk 6]
```diff
@@ -62,6 +62,65 @@ def list_wallet_ids
...
+
+      existing_seeds_in_store[data.wallet_id] = {
+        seed: seed_to_store,
+        encrypted: encrypt_flag,
+        auth_tag: auth_tag,
+        iv: iv,
+      }
+
+      File.open(Coinbase.configuration.backup_file_path, 'w') do |file|
+        file.write(JSON.pretty_generate(existing_seeds_in_store))
```
Comment (yuga-cb) [inline lib/coinbase/user.rb:94]: Does this overwrite the existing file?

File: lib/coinbase/user.rb (hunk 5 above)
Comment (yuga-cb) [reply lib/coinbase/user.rb:66]: There may also be some language missing about the behavior of what happens when you call save twice (i.e. does it overwrite, or append?)

File: spec/unit/coinbase/user_spec.rb [hunk 7]
```diff
@@ -127,4 +127,134 @@
...
+      expect(addresses_api).to receive(:get_address).and_return(address_model)
+      File.open(Coinbase.configuration.backup_file_path, 'w') do |file|
+        file.write(JSON.pretty_generate(initial_seed_data))
+      end
+    end
+    after do
+      File.delete(Coinbase.configuration.backup_file_path)
+    end
+
+    it 'loads the Wallet from backup' do
```
Comment (yuga-cb) [inline spec/unit/coinbase/user_spec.rb:252]: Can we add cases for no backup file, corrupted backup file?

**Comment by yuga-cb** [comment]
I think u.save_wallet and u.load_wallets is the right naming here, to make it consistent with create_wallet

File: spec/unit/coinbase/user_spec.rb (hunk 7 above)
Comment (jazz-cb) [reply spec/unit/coinbase/user_spec.rb:301]: added

File: lib/coinbase/user.rb (hunk 5 above)
Comment (jazz-cb) [reply lib/coinbase/user.rb]: updated. since the function takes wallet_model itself, we dont need to add details on sessions i think

**Review by yuga-cb — APPROVED** [review]

**Pull request merged**